### DISCUSSION TITLE
Total Novice - SV06 Plus to Klipper

### DISCUSSION THREAD

**Discussion by Morrowshadow** [bassamanator/Sovol-SV06-firmware#186] [Q&A]
Hello, so today my SV06 Plus has finally arrived! I'm looking to install Klipper to it, but I'm afraid my knowledge is rather limited, but I'd love to learn.

Can klipper be put on the SV06 Plus' stock board? I'm not bothered about losing the screen as I usually use fluidd anyway for my neptune 4 pro.

I don't have a raspberry pi or any BTT boards or anything. I can order some if needed, but I'd like guidance first. Please dumb it right down for me :')

**Comment by bassamanator** [comment]
> Can klipper be put on the SV06 Plus' stock board? I'm not bothered about losing the screen as I usually use fluidd anyway for my neptune 4 pro.

Yes.

> I don't have a raspberry pi or any BTT boards or anything. I can order some if needed, but I'd like guidance first. Please dumb it right down for me :')

JFYI, for klipper to work, you need a printer motherboard flashed with klipper firmware, and a (linux) computer of some kind that will host klipper (this is the brains of the operation). Any old computer, laptop, potato, etc., can host klipper. Used mini pcs are great for this, you don't need anything fancy.

Read the documentation on this repo fully, sv06-plus branch.

**Reply by Morrowshadow** [reply]
Ah, so I'd need something else hardware wise other than the stock sv06 plus mainboard?

Do you have any suggestions? I sadly don't have any old mini-pc's or a spare PC to host.

I was hoping to make the printer as neat as possible 😄 I do also plan on adding the cartographer (beacon clone) so klipper is essential

**Comment by bassamanator** [comment]
I have a preference for Orange Pi devices. My Opi Zero 2W (CAD$28) can likely host 3 or more printers, but I would need to employ a usb hub.

You just need to know that you don't need a powerful computer to host klipper, even a Raspberry pi zero, which is a very lower powered (and overpriced), can host klipper. If you have the money to spare, buy yourself a Rpi 4 of any kind, and mount it on your printer using my mounting solution (printables). This way you won't be limited by port count.

**Reply by Morrowshadow** [reply]
Thank you so much for the advice! 

I'm assuming RAM size doesn't really matter for this usecase? The host will only be connected to the SV06+, and I have usb wifi adapters and hubs to spare if they're needed!

**Reply by bassamanator** [reply]
The rpi 4 already has built in wifi. The ram size doesn't matter, I think the lowest you can get is 2gb, buy what you can afford.

**Reply by Morrowshadow** [reply]
Will keep an eye out!

Reason I asked about a RPi 3 B+ is because there's one local to me for £25

**Reply by bassamanator** [reply]
The rpi 3 will be just fine. It's just that out of my 4 rpis, I like the rpi 3 the least.

Go for the rpi 3 as long as the price is right.

**Comment by Morrowshadow** [comment]
So i managed to grab a RPi Model 4 B and i've done all the steps. sadly it looks like it still isn't communicating with the printer. Mainsail just says the following: 
![Screenshot 2024-10-03 155757](https://github.com/user-attachments/assets/2b7d205b-d8d1-4674-8a4d-00447e3158a4)

**Comment by bassamanator** [comment]
If you follow the manual, you won't encounter such issues. Please read the manual.
